pkg/operator/controller/externaldns/pod.go

```diff
@@ -170,7 +171,58 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 
 	//TODO: Add logic for the CRD source.
 
+	filterArgs, err := b.domainFilters()
+	if err != nil {
+		return err
+	}
+	container.Args = append(container.Args, filterArgs...)
 	container.Args = append(container.Args, args...)
+	return nil
+}
+
+func (b *externalDNSContainerBuilder) domainFilters() ([]string, error) {
+	var (
+		regexDomainIncludeSet bool
+		regexDomainExcludeSet bool
+		args                  []string
+	)
+	for _, d := range b.externalDNS.Spec.Domains {
+		switch d.FilterType {
+		case operatorv1alpha1.FilterTypeInclude:
+			switch d.MatchType {
+			case operatorv1alpha1.DomainMatchTypeExact:
+				if d.Name == nil {
+					return nil, fmt.Errorf("name for domain cannot be empty")
+				}
+				args = append(args, fmt.Sprintf("--domain-filter=%s", *d.Name))
+			case operatorv1alpha1.DomainMatchTypeRegex:
+				if d.Pattern == nil {
+					return nil, fmt.Errorf("pattern for domain cannot be empty")
+				}
+				if regexDomainIncludeSet {
+					return nil, fmt.Errorf("more than one domain pattern cannot be specified for inclusion")
+				}
+				args = append(args, fmt.Sprintf("--regex-domain-filter=%s", *d.Pattern))
```

Comment:
Just noticed that `--regexp-*` flags were added in version `0.8.0` ([link](https://github.com/kubernetes-sigs/external-dns/releases/tag/v0.8.0)) and we are still using `0.7.1` ([link](https://github.com/openshift/external-dns-operator/blob/main/pkg/operator/config/config.go#L21)).

I think it's fine to upgrade to `0.8.0` in our code but it would be good to test it at least manually.

Reply:
Hmm, then I wonder why it was added to the CRD. I will update the external-dns version.

Reply:
Updated.

Reply:
Yeah, I put the same version as in our downstream fork of external-dns but it seems that the API was updated with the regexp filters after the fork was created.